lib/nostrum/struct/embed.ex

```diff
@@ -381,6 +422,46 @@ defmodule Nostrum.Struct.Embed do
     put_field(%__MODULE__{embed | fields: []}, name, value, inline)
   end
 
+  @doc """
+  Create an embed from a struct that implements the protocol `Nostrum.Struct.Embed.Protocol`.
```

Comment:
I don't think I'm 100% happy with the wording here. Structs don't necessarily implement a protocol per se, right? It's the module that implements the behavior. I obviously get what you're trying to say but it's not 100% correct and if someone is unfamiliar with behaviors I could see there being issues.

Also, this still mentions a protocol when we're using a behavior now.

Reply:
My bad. I wrote this while I still relied on protocols and forgot to update it 🙏